### ISSUE TITLE
DataSet and MultiDataSet merge methods: add additional checks

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/nd4j#2670]
> Tim Hamann @hamannTim Feb 26 22:41
Hi there, I'm building an LSTM (many-to-one) with a FileDataSetIterator that @AlexDBlack provided me (I think).
today, I got this exception when calling model.fit(iter): https://gist.github.com/hamannTim/2e749e8f9b0b9e8b3d3074b2089966c6

I suspect this is *probably* due to an empty DataSet or MultiDataSet (i.e., missing features/labels arrays)... this should be checked and (where appropriate) handled - or at least a useful/informative exception should be thrown.
